src/app/pages/catalog/catalog.component.html

```diff
@@ -9,8 +9,8 @@ <h1 class="mt-20 text-center">
     </span>
   </h1>
   <div class="m-4 flex flex-wrap items-center justify-center gap-6">
-    @for (quizCategoryData of categories; track $index) {
-      <quiz-quiz-card [quizCategory]="quizCategoryData"></quiz-quiz-card>
+    @for (category of categories$ | async; track category.id) {
```

Comment:
let's say there is a failed get request, you got an empty array here, right? How your app should behave in this case? Think about content that should be hidden on this page, maybe a message to a user.

Reply:
Also while getting categories, we should display the pending progress to a user by adding a spinner or skeleton.

Reply:
> you got an empty array here, right?

Right, it's my temporary solution.
I'l think about better solution. I will add two tickets to the trello board : add user message ,  add spinner. It will be Implemented  in  separated PR's.